### PR TITLE
Unignore fastcdr when building for connext.

### PR DESCRIPTION
**Pull request by clalancette**
The new Connext RMW relies on fastcdr to do generation for
it, so it won't build without it.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

### CODE REVIEW

File: rolling/ci-nightly-connext.yaml [hunk 1]
```diff
@@ -26,7 +26,7 @@ jenkins_job_priority: 50
 jenkins_job_schedule: 15 23 * * *
 jenkins_job_timeout: 300
 jenkins_job_weight: 4
-package_selection_args: '--packages-ignore fastcdr foonathan_memory_vendor rosbag2_converter_default_plugins --packages-ignore-regex .*cyclonedds.* .*fastrtps.*'
+package_selection_args: '--packages-ignore foonathan_memory_vendor rosbag2_converter_default_plugins --packages-ignore-regex .*cyclonedds.* .*fastrtps.*'
```
Comment (cottsay) [inline rolling/ci-nightly-connext.yaml]: Doesn't `rmw_connextdds` require the rosidl packages for Fast-RTPS as well?

Comment (clalancette) [reply rolling/ci-nightly-connext.yaml]: Oh, I missed that in the regex.  Good call, I'll remove that as well.

File: rolling/ci-nightly-connext.yaml [hunk 2]
```diff
@@ -26,7 +26,7 @@ jenkins_job_priority: 50
 jenkins_job_schedule: 15 23 * * *
 jenkins_job_timeout: 300
 jenkins_job_weight: 4
-package_selection_args: '--packages-ignore fastcdr foonathan_memory_vendor rosbag2_converter_default_plugins --packages-ignore-regex .*cyclonedds.* .*fastrtps.*'
+package_selection_args: '--packages-ignore foonathan_memory_vendor rosbag2_converter_default_plugins --packages-ignore-regex .*cyclonedds.* rmw_fastrtps.*'
```
Comment (cottsay) [inline rolling/ci-nightly-connext.yaml]: Still want to ignore `fastrtps` itself:
```suggestion
package_selection_args: '--packages-ignore fastrtps foonathan_memory_vendor rosbag2_converter_default_plugins --packages-ignore-regex .*cyclonedds.* rmw_fastrtps.*'
```

**Comment by clalancette** [comment]
It turns out that we can't ignore fastrtps, since the rosidl_typesupport_fastrtps packages depend on it.  In turn, that means we also cannot ignore foonathan_memory_vendor, which fastrtps depends on.  So I'll remove both of those from the `--packages-ignore` list.

**Comment by clalancette** [comment]
Here's a build using the current state of this PR: https://build.ros2.org/view/Rci/job/Rci__nightly-connext_ubuntu_focal_amd64/273

**Review by asorbini — APPROVED** [review]
LGTM.

Hopefully we can remove the dependency on `fastrtps` from `rosidl_typesupport_fastrtps` in the future. I asked @MiguelCompany about it ([here](https://github.com/ros2/rmw_connextdds/pull/14#issuecomment-809552634)), and I can open a PR on that repository if desired (and feasible... I will wait to hear back).

**Comment by clalancette** [comment]
I'm going to go ahead and merge this to get this job at least yellow on the next nightlies.  If there is any other review feedback, I'm happy to address it afterwards.

**Pull request merged**